hub/package-manager/winget/pinning.md

```diff
@@ -0,0 +1,234 @@
+
+WinGet pinning supports three types of Package Pinning:
+- **Pinning**:
+The package is excluded from winget upgrade --all but allows winget upgrade <WinGet package>. You can use the "–include-pinned” argument to let winget upgrade --all include pinned packages.
+
+- **Blocking**:
+The package is blocked from winget upgrade --all or winget upgrade <WinGet package>, you will have to unblock the package to let WinGet perform an upgrade.
+
+- **Gating**:
+The package is pinned to specific version(s). For example, if a package is pinned to version 1.2.*, any version between 1.2.0 to 1.2.x is considered valid. To allow user override, “–force” can be used with winget upgrade <WinGet package> to override some of the pinning created above.
+
+## Usage
+
+```cmd
+winget pin <subcommand> <options>
+```
+## Arguments
+
+The following arguments are available.
+
+| Argument  | Description |
+|--------------|-------------|
+| **-?, --help** |  Gets additional help on this command. |
+
+## Subcommands
+
+The **pin** command supports the following subcommands.
+
+| Subcommand  | Description |
+|--------------|-------------|
+|  **add** |  Adds a new pin. |
+|  **remove** | Remove a package pin. |
+|  **list** | List current pins. |
+|  **reset** | Reset pins |
+
+## add
+
+The **add** subcommand adds a new pin. This subcommand requires that you specify the exact package to pin. If there is any ambiguity, you will be prompted to further filter the **add** subcommand to an exact application.
+
+Usage:
+
+```cmd
+winget pin add [[-q] <query>] [<options>]
+```
+
+### Arguments
+
+| Argument      | Description |
+|-------------|-------------|  
+| **-q,--query**  |  The query used to search for an app. |
+| **-?, --help** |  Get additional help on this command. |
+
+### Options
+
+The options allow you to customize adding pins to meet your needs.
+
+| Option  | Description |
+|-------------|-------------|  
+| **--id**    |  Limits the search to the ID of the application.   |  
+| **--name**   |  Limits the search to the name of the application. |  
+| **--moniker**   | Limits the search to the moniker listed for the application. |  
+| **--tag**   | Limits the search to the tag listed for the application. | 
+| **--cmd, --command**   | Limits the search to the command of the application. |   
+| **-v, --version**  |  Enables you to specify an exact version to pin. The wildcard '*' can be used as the last version part. |  
+| **-s, --source**   |  Restricts the search to the source name provided. Must be followed by the source name. |  
+| **-e, --exact**   |   Uses the exact string in the query, including checking for case-sensitivity. It will not use the default behavior of a substring. |  
+| **--force** | Overrides the installer hash check. Not recommended. |
```

Comment:
```suggestion
| **--force** | Direct run the command and continue with non security related issues |
```
This used to be the old behavior and I believe `--ignore-security-hash` was added to `winget install` / `winget upgrade` instead for this purpose. I copied this description from Windows Package Manager v1.5.1572 description of` winget pin add` command.